tensorboard/webapp/customization/customization_test.ts

```diff
@@ -0,0 +1,113 @@
+/* Copyright 2021 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+import {Component, NgModule, Optional} from '@angular/core';
+import {TestBed} from '@angular/core/testing';
+
+import {CustomizationModule} from './customization_module';
+
+/**
+ * Marker class for the component that can be customized in these tests.
+ * Customizations should implement this type and be provided.
+ */
+export class CustomizableComponentType {}
+
+/**
+ * Parent class that uses the <tb-customization> component.
+ */
+@Component({
+  selector: 'parent-component',
+  template: `
+    <tb-customization [customizableComponent]="customizableComponent">
+      <div class="default">Showing Default Text!</div>
+    </tb-customization>
+  `,
+})
+export class ParentComponent {
+  constructor(
+    @Optional() readonly customizableComponent: CustomizableComponentType
+  ) {}
+}
+
+/**
+ * Imports CustomizationModule for ParentComponent.
+ */
+@NgModule({
+  imports: [CustomizationModule],
+  declarations: [ParentComponent],
+  entryComponents: [ParentComponent],
+})
+export class ParentComponentModule {}
+
+/**
+ * An implementation of CustomizableComponentType to be provided and injected
+ * into the ParentComponent for some tests.
+ */
+@Component({
+  selector: 'customizable-component',
+  template: ` <div>Showing Customized Text!</div> `,
+})
+export class CustomizableComponent implements CustomizableComponentType {}
```

Comment:
`implements CustomizableComponentType` part is a bit odd even that it is not an interface and we really don't want to support invoking method on a component. Just omit it to prevent others from getting a wrong idea?

Reply:
Removed.